hal/src/hal-sdcard.adb

```diff
@@ -0,0 +1,382 @@
+with Ada.Text_IO; use Ada.Text_IO;
```

Comment:
right, maybe simplify it to sd.

On the services side, it's an abstract implementation of the initialisation protocol defined in the standard. I'm not sure it's service as it's support for drivers implementation (that would be a mix of layers: the service layer to me relies on drivers, not really the other way around)